PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/DatabaseResource.java

```diff
@@ -89,7 +89,8 @@ default boolean write(@Nonnull final Element resourceEle, @Nonnull final Databas
             resourceEle.addContent(new Element("url").setText(resource.jdbcUrl.toString()));
             resourceEle.addContent(new Element("username").setText(resource.username));
             resourceEle.addContent(new Element("passwordSave").setText(resource.password.saveType().name()));
-            if (ArrayUtils.isNotEmpty(resource.password.password())) {
+            if (ArrayUtils.isNotEmpty(resource.password.password()) && !StringUtils.equals(String.valueOf(resource.password.password()),
+                    PasswordStore.loadPassword(resource.getType(), resource.getId(), resource.getUsername(), resource.password.saveType()))) {
```

Comment:
can you extract some variables to make it more readable?
```
final String oldPassword = PasswordStore.loadPassword(resource.getType(), resource.getId(), resource.getUsername(), resource.password.saveType());
final String newPassword = String.valueOf(resource.password.password());
if(StringUtils.isNotEmpty(newPassword) && !ObjectUtils.equals(newPassword, oldPassword)){
  //...
}
```

Reply:
what if `newPassword` is empty? will `newPassword` be empty?

Reply:
updated.

Reply:
`newPassword ` is never empty. these is a condition `ArrayUtils.isNotEmpty(resource.password.password())`